controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -112,47 +106,16 @@ func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Con
 	findImageName := re.FindStringSubmatch(servingRuntimeImage)
 	servingRuntime := findImageName[1]
 
-	var data []byte
+	var data string
 	switch servingRuntime {
```

Comment:
(optional)  This is not a blocker. just for recommendation.

You changed the json to constant so the runtime name and also metricsData are String. If so, I think using map would be easier to handle this flow. something like here:
~~~
	runtimeMetricsData := map[string]string{
		constants.OvmsImageName:   constants.OvmsData,
		constants.TgisImageName:   constants.TgisData,
		constants.VllmImageName:   constants.VllmData,
		constants.CaikitImageName: constants.CaikitData,
	}

	data, supported := runtimeData[servingRuntime]
~~~
if no supported runtime comes, it will return the following:
```
data = ""  //(empty string)
supported=false
```

with this, you can use a unified creating ConfigMap process. 
this is just example(not tested)
~~~
	configMap := &corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name:      metricsCMName,
			Namespace: isvc.Namespace,
		},
		Data: map[string]string{
			"supported": supported,
		},
	}
	if supported == "true" {
		configMap.Data["metrics"] = metrics
	}

	if err := ctrl.SetControllerReference(isvc, configMap, r.client.Scheme()); err != nil {
		log.Error(err, "Unable to add OwnerReference to the Metrics Dashboard Configmap")
		return nil, err
	}
~~~